hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/s3guard/ITestDynamoDBMetadataStore.java

```diff
@@ -274,7 +300,9 @@ public void testInitialize() throws IOException {
         getTestTableName("testInitialize");
     final Configuration conf = s3afs.getConf();
     conf.set(S3GUARD_DDB_TABLE_NAME_KEY, tableName);
-    try (DynamoDBMetadataStore ddbms = new DynamoDBMetadataStore()) {
+    enableOnDemand(conf);
```

Comment:
You add this in @BeforeClass as well. Is there a point to add this to all tests as well? Maybe add this to `@Before` only?

Reply:
the beforeClass is for the static ddbms; the others were for the new ones

Reply:
I'll be doing it in the FS configuration and the one for the static metastore -they are different configurations